src/cljs/main/broadfcui/page/method_repo/method/details.cljs

```diff
@@ -197,7 +217,14 @@
                    (fn [{:keys [success? parsed-response]}]
                      (if success?
                        (swap! state assoc :selected-snapshot parsed-response :loading-snapshot? false)
-                       (swap! state assoc :method-error (:message parsed-response)))))})))})
+                       (swap! state assoc :method-error (:message parsed-response)))))})
+       (endpoints/call-ajax-orch
+        {:endpoint (endpoints/get-agora-entity-acl false {:namespace namespace :name name :snapshotId snapshot-id})
+         :on-done (net/handle-ajax-response
+                   (fn [{:keys [success? parsed-response]}]
+                     (if success?
+                       (swap! state assoc :is-public (= (:role (first (filter #(= "public" (:user %)) parsed-response))) "READER"))
+                       (swap! state assoc :is-public false))))})))})
```

Comment:
I do think this needs to move into the back end, with agora returning a "public:true" flag as appropriate. Most importantly - this ajax call will only succeed if you are owner on the snapshot. If you only have reader - and for public methods, most of the world will only be readers - this results in a 403, and we won't display the public icon.

Reply:
I agree with David- this shouldn't be a frontend check...